simulation/amaru-sim/src/simulator/simulate.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Spawn one pipeline for each node in the test;
```

Comment:
Perhaps worthwhile to turn those comments into proper rust documentation? I think it would be great to start publishing our doc somewhere to share with others